src/Language/Fixpoint/Types/Names.hs

```diff
@@ -44,6 +44,7 @@ module Language.Fixpoint.Types.Names (
   , dropSym
   , headSym
   , lengthSym
+  , mappendSym
```

Comment:
I see, it has to do with overlapping instances. Why do you need to create the `Symbol` yourself then? Are you trying to lookup the dictionary `Id` by recreating the name? There must be a better way.